formats/json/commonMain/src/kotlinx/serialization/json/internal/JsonStreams.kt

```diff
@@ -48,8 +52,12 @@ internal fun <T> Json.decodeToSequenceByReader(
     format: DecodeSequenceMode = DecodeSequenceMode.AUTO_DETECT
 ): Sequence<T> {
     val lexer = ReaderJsonLexer(reader)
-    val iter = JsonIterator(format, this, lexer, deserializer)
-    return Sequence { iter }.constrainOnce()
+    try {
+        val iter = JsonIterator(format, this, lexer, deserializer)
+        return Sequence { iter }.constrainOnce()
+    } finally {
+        lexer.release()
```

Comment:
I may be wrong, but we can't do `release` here because the sequence is not yet evaluated, so a buffer is still needed

Reply:
I'll double-check, thanks

Reply:
You a right: added stress test for each mode and fixed the issue